installer/pkg/components/image-builder-mk3/deployment.go

```diff
@@ -22,24 +19,24 @@ import (
 	"k8s.io/utils/pointer"
 )
 
-// todo(sje): work out what to do here - this may be better in common, or to autogenerate on everything
-func getChecksum(path string) string {
-	hash := sha256.Sum256([]byte(path))
-
-	return hex.EncodeToString(hash[:])
-}
-
 func deployment(ctx *common.RenderContext) ([]runtime.Object, error) {
 	labels := common.DefaultLabels(Component)
 
-	annotations := map[string]string{
-		// todo(sje): put the configmap
-		"checksum/image-builder-mk3-configmap": getChecksum("image-builder-mk3-configmap.yaml"),
+	var hashObj []runtime.Object
+	if objs, err := configmap(ctx); err != nil {
+		return nil, err
+	} else {
+		hashObj = append(hashObj, objs...)
+	}
+	if objs, err := secret(ctx); err != nil {
```

Comment:
Strictly speaking, this isn't going to be the correct hash - the `secret` function generates the secret each time it's run. What I'd propose is removing the `secret` (and probably `configmap`) call from the `objects.go` file and export them from the `deployment.go` file.

It would also make it a bit more efficient as the functions don't need to be generated twice then